File: Data Structures & Algorithms/eating-bananas/submission-3.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) {
        let left = 1;
        let right = Math.max(...piles);
        let res = right;

        while (left <= right) {
            console.log(left,right,res);
            let mid = Math.floor((left+right)/2);
            let hours = 0;

            for (let pile of piles) {
                hours += Math.ceil(pile/mid);
            }

            if (hours <= h) {
                res = Math.min(res, mid);
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        largest = 0
        left, right = 0, len(heights)-1

        while left < right:
            height = min(heights[left], heights[right])
            largest = max(largest, (right-left)*height)

            if heights[left] > heights[right]:
                right -= 1
            else:
                left += 1
        
        return largest